manifests/plugin/supervisory_authority.pp

```diff
@@ -42,6 +42,21 @@
   String                       $metrics_interval      = '30s',
 ) {
   foreman::plugin { 'supervisory_authority':
-    config => template('foreman/foreman_supervisory_authority.yaml.erb'),
+    config => epp(
+      'foreman/foreman_supervisory_authority.yaml.epp',
+      {
+        server_url            => $server_url,
+        secret_token          => $secret_token,
+        service_name          => $service_name,
+        log_level             => $log_level,
+        pool_size             => $pool_size,
+        api_buffer_size       => $api_buffer_size,
+        api_request_size      => $api_request_size,
+        api_request_time      => $api_request_time,
+        transaction_max_spans => $transaction_max_spans,
+        http_compression      => $http_compression,
+        metrics_interval      => $metrics_interval,
+      }
+    ),
```

Comment:
Untested, but I think it can actually not use any template and use this instead:
```suggestion
    config => foreman::to_symbolized_yaml(
      {
        server_url            => $server_url,
        secret_token          => $secret_token,
        service_name          => $service_name,
        log_level             => $log_level,
        pool_size             => $pool_size,
        api_buffer_size       => $api_buffer_size,
        api_request_size      => $api_request_size,
        api_request_time      => $api_request_time,
        transaction_max_spans => $transaction_max_spans,
        http_compression      => $http_compression,
        metrics_interval      => $metrics_interval,
      }
    ),
```

Reply:
Maybe You can test it later, if the Function can handle `Sensitive[String] $secret_token`.

Reply:
You don't need to because you can unwrap when passing the hash.

Reply:
I thought, this was the whole Point of the Story: Not having to spread random `unwrap`s in the Code.  If the Function `foreman::to_symbolized_yaml` cannot handle `Sensitive`, then it has to be empowered to do so, and just like the Function `epp` it should return `Sensitive` then.

Reply:
Besides that, `foreman::to_symbolized_yaml` generates a symbolized(!) YAML like… 
```
---
:a: A
:b:
  x: X
  y: Y
:c:
- 1
- 2
- 3

```
…  (note the leading `:`)  I guess that's not, what the Configfile wants.

Reply:
Have a look at the template: it's exactly what it wants.

Reply:
Yes, You are right 👍   

[I upgraded Your Function](https://github.com/cocker-cc/puppet-foreman/compare/6c902a4f1f484137ebccafe4d3390881fd0b3d61..1fba024a8d226a0d7c77822c09bce363bc107f96) to be sensitive-aware.  
What do You think of a Pullrequest?

Reply:
For our use case right now I think it only complicates matters. I'd prefer to unwrap it in Puppet code.

Reply:
If You want to, save my Code somewhere.  I will delete it, after this Story's End.